compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -1830,22 +1830,27 @@ private void loadInvokableType(MethodVisitor mv, BInvokableType bType) {
         mv.visitTypeInsn(NEW, FUNCTION_TYPE_IMPL);
         mv.visitInsn(DUP);
 
-        // Create param types array
-        mv.visitLdcInsn((long) bType.paramTypes.size());
-        mv.visitInsn(L2I);
-        mv.visitTypeInsn(ANEWARRAY, TYPE);
-        int i = 0;
-        for (BType paramType : bType.paramTypes) {
-            mv.visitInsn(DUP);
-            mv.visitLdcInsn((long) i);
+        if (!Symbols.isFlagOn(bType.flags, Flags.ANY_FUNCTION)) {
+            // Create param types array
+            mv.visitLdcInsn((long) bType.paramTypes.size());
             mv.visitInsn(L2I);
+            mv.visitTypeInsn(ANEWARRAY, TYPE);
+            int i = 0;
+            for (BType paramType : bType.paramTypes) {
+                mv.visitInsn(DUP);
+                mv.visitLdcInsn((long) i);
+                mv.visitInsn(L2I);
 
-            // load param type
-            loadType(mv, paramType);
+                // load param type
+                loadType(mv, paramType);
 
-            // Add the member to the array
-            mv.visitInsn(AASTORE);
-            i += 1;
+                // Add the member to the array
+                mv.visitInsn(AASTORE);
+                i += 1;
+            }
+        } else {
+            mv.visitLdcInsn(0);
+            mv.visitTypeInsn(ANEWARRAY, TYPE);
```

Comment:
If we do this, the runtime representation changes from that at compile-time, right?

For `function`, at compile time the param types and return type will be null. But if we do this, at runtime, param types will be an empty array and return type will be nil (I believe from L1864).

I suggest we modify the `io.ballerina.runtime.internal.types.BFunctionType#BFunctionType()` constructor in `BFunctionType` to represent this `function` type. It currently seems to be used only as the default constructor in `io.ballerina.runtime.internal.types.BMethodType#BMethodType`, but I think we can modify that a bit to call a different constructor in `BFunctionType`.

@warunalakshitha, thoughts please? To provide some context, `function` represents any function in Ballerina, and doesn't have param or return types.

```ballerina
function anyFunc = main;

public function main() {
 
}
```

Reply:
Fixed as we discussed offline.